client/main.lua

```diff
@@ -1,274 +1,216 @@
+        lib.addRadialItem({
+            id = 'emergencybutton2',
+            label = Lang:t("options.emergency_button"),
+            icon = 'circle-exclamation',
+            onSelect = function (_, _)
+                if ispolice then
+                    TriggerEvent('police:client:SendPoliceEmergencyAlert')
+                elseif isems then
+                    TriggerEvent('ambulance:client:SendAmbulanceEmergencyAlert')
+                end
+                lib.hideRadial()
+            end
+        })
     else
-        SetupSubItems()
-        FinalMenuItems = deepcopy(Config.MenuItems)
-        for _, v in pairs(DynamicMenuItems) do
-            FinalMenuItems[#FinalMenuItems+1] = v
-        end
-
-    end
-end
-
-local function setRadialState(bool, sendMessage, delay)
-    -- Menuitems have to be added only once
-
-    if bool then
-        TriggerEvent('qb-radialmenu:client:onRadialmenuOpen')
+        lib.clearRadialItems()
         SetupRadialMenu()
-    else
-        TriggerEvent('qb-radialmenu:client:onRadialmenuClose')
+        lib.disableRadial(false)
     end
-
-    SetNuiFocus(bool, bool)
-    if sendMessage then
-        SendNUIMessage({
-            action = "ui",
-            radial = bool,
-            items = FinalMenuItems
-        })
-    end
-    if delay then Wait(500) end
-    inRadialMenu = bool
-end
-
--- Command
-
-RegisterCommand('radialmenu', function()
-    if ((IsDowned() and IsPoliceOrEMS()) or not IsDowned()) and not PlayerData.metadata["ishandcuffed"] and not IsPauseMenuActive() and not inRadialMenu then
-        setRadialState(true, true)
-        SetCursorLocation(0.5, 0.5)
-    end
-end)
-
-RegisterKeyMapping('radialmenu', Lang:t("general.command_description"), 'keyboard', 'F1')
-
--- Events
-
--- Sets the metadata when the player spawns
-RegisterNetEvent('QBCore:Client:OnPlayerLoaded', function()
-    PlayerData = QBCore.Functions.GetPlayerData()
 end)
 
--- Sets the playerdata to an empty table when the player has quit or did /logout
-RegisterNetEvent('QBCore:Client:OnPlayerUnload', function()
-    PlayerData = {}
+RegisterNetEvent('radialmenu:client:ChangeSeat', function(id, label)
+    local Veh = GetVehiclePedIsIn(cache.ped, false)
+    local IsSeatFree = IsVehicleSeatFree(Veh, id - 2)
+    local speed = GetEntitySpeed(Veh)
+    local HasHarness = exports['qb-smallresources']:HasHarness()
+    if not HasHarness then
+        local kmh = speed * 3.6
+        if IsSeatFree then
+            if kmh <= 100.0 then
+                SetPedIntoVehicle(cache.ped, Veh, id - 2)
+                QBCore.Functions.Notify(Lang:t("info.switched_seats", {seat = label}))
+            else
+                QBCore.Functions.Notify(Lang:t("error.vehicle_driving_fast"), 'error')
+            end
+        else
+            QBCore.Functions.Notify(Lang:t("error.seat_occupied"), 'error')
+        end
+    else
+        QBCore.Functions.Notify(Lang:t("error.race_harness_on"), 'error')
+    end
```

Comment:
```suggestion
    if HasHarness then
        QBCore.Functions.Notify(Lang:t("error.race_harness_on"), 'error')
        return
    end
    
    if not IsSeatFree then
        QBCore.Functions.Notify(Lang:t("error.seat_occupied"), 'error')
        return
    end

    local kmh = speed * 3.6

    if kmh <= 100.0 then
        SetPedIntoVehicle(cache.ped, Veh, id - 2)
        QBCore.Functions.Notify(Lang:t("info.switched_seats", {seat = label}))
    else
        QBCore.Functions.Notify(Lang:t("error.vehicle_driving_fast"), 'error')
    end
```

Restructuring this with guard clauses allows clear and concise requirement checks (IsSeatFree and HasHarness).